staging/src/k8s.io/apimachinery/pkg/util/yaml/decoder.go

```diff
@@ -92,6 +92,8 @@ type YAMLDecoder struct {
 // the caller in framing the chunk.
 func NewDocumentDecoder(r io.ReadCloser) io.ReadCloser {
 	scanner := bufio.NewScanner(r)
+	buf := make([]byte, 64*1024)
+	scanner.Buffer(buf, 5*1024*1024)
```

Comment:
how was 5MB selected as a target?

Reply:
https://github.com/kubernetes/kubernetes/issues/89016#issuecomment-597340082

Random thought from me. My thinking was this is generally client-side, and done ~once per client utility run.

But I agree with the suggestion of a smallish buffer paired with a large max size, that's better.